### DISCUSSION TITLE
S3 Support

### DISCUSSION THREAD

**Discussion by JJGadgets** [go-shiori/shiori#798] [Feature requests]
It would be nice to have support for storing the data stated in the Storage docs on an S3 provider (e.g. Ceph RGW, MinIO, AWS S3, Wasabi, Cloudflare R2, etc).

My current use case: I'd like to avoid using any Kubernetes PVCs for Shiori's pod, to avoid mounting issues (Ceph, NFS, etc). Since Shiori can already consume an external database like Postgres for metadata storage, I would also like to have Shiori consume external S3 for data storage natively.

**Comment by fmartingr** [comment]
In one of the PRs I have open I'm slowly switching to a more agnostic filesystem implementation that would (theoretically) support other filesystems. Right now this is not on my radar, but if we manage to refactor and test all instances of filesystem-aware logic this should be easy to implement for new installations, though we would need to properly document how to migrate the data and so on.
